src/generators/tracingFileGenerator.ts

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Project, SourceFile } from "ts-morph";
+import { ClientDetails } from "../models/clientDetails";
+import { PackageDetails } from "../models/packageDetails";
+import { NameType, normalizeName } from "../utils/nameUtils";
+import {
+  generateOperationJSDoc,
+  ParameterWithDescription
+} from "./utils/docsUtils";
+
+export function generateTracingFile(
+  clientDetails: ClientDetails,
+  packageDetails: PackageDetails,
+  project: Project
+) {
+  if (!clientDetails.enableTracing) {
+    return;
+  }
+
+  const file = project.createSourceFile(
+    `${clientDetails.srcPath}/tracing.ts`,
+    undefined,
+    {
+      overwrite: true
+    }
+  );
+
+  file.addImportDeclarations([
+    { namedImports: ["getTracer"], moduleSpecifier: "@azure/core-tracing" },
+    {
+      namedImports: ["Span", "SpanOptions", "SpanKind"],
+      moduleSpecifier: "@opentelemetry/api"
+    },
+    {
+      namedImports: ["OperationOptions"],
+      moduleSpecifier: "@azure/core-http"
+    }
+  ]);
+
+  file.addTypeAlias({
+    name: "OperationTracingOptions",
+    type: `OperationOptions["tracingOptions"];`
+  });
+
+  writeCreateSpanFunction(file, packageDetails);
+}
+
+function writeCreateSpanFunction(
+  file: SourceFile,
+  packageDetails: PackageDetails
+) {
+  const parameters: ParameterWithDescription[] = [
+    {
+      name: "operationName",
+      type: "string",
+      description: "The name of the operation being performed."
+    },
+    {
+      name: "operationOptions",
+      type: "T",
+      description: "The options for the underlying http request."
+    }
+  ];
+
+  const createSpanFunction = file.addFunction({
+    isExported: true,
+    docs: [
+      generateOperationJSDoc(
+        parameters,
+        "Creates a span using the global tracer."
+      )
+    ],
+    name: "createSpan",
+    typeParameters: [
+      {
+        name: "T",
+        constraint: "OperationOptions"
+      }
+    ],
+    parameters,
+    returnType: "{ span: Span; updatedOptions: T }"
+  });
+
+  const telemetryNamespace = getTelemetryNamespace(packageDetails);
+  const telemetryPackageName = `Microsoft.${getTelemetryPackageName(
+    packageDetails
+  )}`;
+  createSpanFunction.addStatements([
```

Comment:
I've been thinking for a while that we could move most of this helper into `core-tracing` itself and then do something like

```ts
import { createSpanFunction } from "@azure/core-tracing";
const createSpan = createSpanFunction({packagePrefix, namespace});
```

thoughts?

Reply:
Yeah totally makes sense added it to `core-http` and `core-client` [#12525](https://github.com/Azure/azure-sdk-for-js/pull/12525)